backend/addcorpus/save_corpus.py

```diff
@@ -0,0 +1,131 @@
+    '''
+    Save a corpus configuration in the SQL database.
+
+    Parameters:
+    - `corpus`: a Corpus database object, which should not have an existing configuration
+    - `corpus_definition`: a corpus object, output of `load_corpus`
+    '''
+
+    configuration = CorpusConfiguration(corpus=corpus)
+    _copy_corpus_attributes(corpus_definition, configuration)
+    configuration.save()
+    configuration.full_clean()
+
+    _save_corpus_fields_in_database(corpus_definition, configuration)
+
+def get_defined_attributes(object, attributes):
+    get = lambda attr: object.__getattribute__(attr)
+    has_attribute = lambda attr: attr in dir(object) and get(attr) != None
+
+    return {
+        attr: get(attr)
+        for attr in attributes
+        if has_attribute(attr)
+    }
+
+def _copy_corpus_attributes(corpus_definition: CorpusDefinition, configuration: CorpusConfiguration):
+    attributes_to_copy = [
+        'description',
+        'allow_image_download',
+        'category',
+        'description_page',
+        'document_context',
+        'es_alias',
+        'es_index',
+        'image',
+        'languages',
+        'min_date',
+        'max_date',
+        'scan_image_type',
+        'title',
+        'word_models_present',
+    ]
+
+    defined = get_defined_attributes(corpus_definition, attributes_to_copy)
+
+    for attr, value in defined.items():
+        configuration.__setattr__(attr, value)
+
+def _save_corpus_fields_in_database(corpus_definition: CorpusDefinition, configuration: CorpusConfiguration):
+    for field in corpus_definition.fields:
+        _save_field_in_database(field, configuration)
+
+def _save_field_in_database(field_definition: FieldDefinition, configuration: CorpusConfiguration):
+    attributes_to_copy = [
+        'name', 'display_name', 'display_type',
+        'description', 'results_overview',
+        'csv_core', 'search_field_core',
+        'visualizations', 'visualization_sort',
+        'es_mapping', 'indexed', 'hidden',
+        'required', 'sortable', 'primary_sort',
+        'searchable', 'downloadable'
+    ]
+
+    copy_attributes = get_defined_attributes(field_definition, attributes_to_copy)
+
+    filter_definition = field_definition.search_filter.serialize() if field_definition.search_filter else {}
+
+    field = Field(
+        corpus_configuration=configuration,
+        search_filter=filter_definition,
+        **copy_attributes,
+    )
+
+    field.save()
+    field.full_clean()
+    return field
+
+def _set_corpus_inactive(corpus_name):
+    '''
+    Set a corpus to inactive status, if it exists in the database
+    '''
+
+    exists = Corpus.objects.filter(name=corpus_name).exists()
+
+    if exists:
+        corpus = Corpus.objects.get(name=corpus_name)
+        if corpus.active:
+            corpus.configuration.delete()
+
+def _try_saving_corpus(corpus_name, corpus_definition, verbose = False, stdout=sys.stdout, stderr=sys.stderr):
```

Comment:
The function name "try" does not sit right with me - it raises questions as to what happens when the attempt fails. Perhaps naming this function `_save_or_skip` or similar may make it more explicit what it does?